pydantic/main.py

```diff
@@ -420,6 +420,7 @@ def dict(
         exclude_unset: bool = False,
         exclude_defaults: bool = False,
         exclude_none: bool = False,
+        encode_as_json: bool = False,
```

Comment:
do you really need this on `.dict()`?

Reply:
The function calls go `parent.json()` -> `parent._iter()` -> `parent._get_value()` -> (if child is a BaseModel), `child.dict()` -> `child._iter()` ....

So no, I could alter `_get_value` and call `_iter()` directly. I thought about it but assumed there was a reason to use `dict()` in `_get_value` -- also that maybe having `encode_as_json` in `dict()` could be a feature. Any thoughts?

Reply:
Ok, I understand, best to keep it like this.

Reply:
It may be worth documenting this new parameter. It still looks useful when actual JSON-encoding occurs later.